commit 2b45e05f51a79c2818523c923dfe008b8b2f4227
Author: Jason Cooper <jason@lakedaemon.net>
Date:   Wed Feb 29 17:39:08 2012 +0000

    ARM: kirkwood: fdt: absorb kirkwood_init()
    
    We need to absorb kirkwood_init() into kirkwood_dt_init() so that as we
    convert drivers, we can remove the platform call, eg
    kirkwood_rtc_init().  This maintains compatibility with non-fdt
    configurations because they still call kirkwood_init() in common.c.
    
    As drivers are converted, we will reinstate the 'static' qualifier in
    common.c.
    
    Signed-off-by: Jason Cooper <jason@lakedaemon.net>

Index: sid/arch/arm/mach-kirkwood/board-dt.c
===================================================================
--- sid.orig/arch/arm/mach-kirkwood/board-dt.c	2012-06-10 20:02:17.000000000 +0200
+++ sid/arch/arm/mach-kirkwood/board-dt.c	2012-06-10 20:02:22.677136456 +0200
@@ -29,7 +29,9 @@
 #include <linux/spi/orion_spi.h>
 #include <asm/mach-types.h>
 #include <asm/mach/arch.h>
+#include <asm/mach/map.h>
 #include <mach/kirkwood.h>
+#include <mach/bridge-regs.h>
 #include <plat/mvsdio.h>
 #include "common.h"
 #include "mpp.h"
@@ -155,7 +157,32 @@ static void __init dreamplug_init(void)
 
 static void __init kirkwood_dt_init(void)
 {
-	kirkwood_init();
+	pr_info("Kirkwood: %s, TCLK=%d.\n", kirkwood_id(), kirkwood_tclk);
+
+	/*
+	 * Disable propagation of mbus errors to the CPU local bus,
+	 * as this causes mbus errors (which can occur for example
+	 * for PCI aborts) to throw CPU aborts, which we're not set
+	 * up to deal with.
+	 */
+	writel(readl(CPU_CONFIG) & ~CPU_CONFIG_ERROR_PROP, CPU_CONFIG);
+
+	kirkwood_setup_cpu_mbus();
+
+#ifdef CONFIG_CACHE_FEROCEON_L2
+	kirkwood_l2_init();
+#endif
+
+	/* internal devices that every board has */
+	kirkwood_rtc_init();
+	kirkwood_wdt_init();
+	kirkwood_xor0_init();
+	kirkwood_xor1_init();
+	kirkwood_crypto_init();
+
+#ifdef CONFIG_KEXEC
+	kexec_reinit = kirkwood_enable_pcie;
+#endif
 
 	if (of_machine_is_compatible("globalscale,dreamplug"))
 		dreamplug_init();
Index: sid/arch/arm/mach-kirkwood/common.c
===================================================================
--- sid.orig/arch/arm/mach-kirkwood/common.c	2012-06-10 19:52:45.000000000 +0200
+++ sid/arch/arm/mach-kirkwood/common.c	2012-06-10 20:02:22.677136456 +0200
@@ -164,7 +164,7 @@ void __init kirkwood_nand_init_rnb(struc
 /*****************************************************************************
  * SoC RTC
  ****************************************************************************/
-static void __init kirkwood_rtc_init(void)
+void __init kirkwood_rtc_init(void)
 {
 	orion_rtc_init(RTC_PHYS_BASE, IRQ_KIRKWOOD_RTC);
 }
@@ -282,7 +282,7 @@ void __init kirkwood_crypto_init(void)
 /*****************************************************************************
  * XOR0
  ****************************************************************************/
-static void __init kirkwood_xor0_init(void)
+void __init kirkwood_xor0_init(void)
 {
 	kirkwood_clk_ctrl |= CGC_XOR0;
 
@@ -295,7 +295,7 @@ static void __init kirkwood_xor0_init(vo
 /*****************************************************************************
  * XOR1
  ****************************************************************************/
-static void __init kirkwood_xor1_init(void)
+void __init kirkwood_xor1_init(void)
 {
 	kirkwood_clk_ctrl |= CGC_XOR1;
 
@@ -307,7 +307,7 @@ static void __init kirkwood_xor1_init(vo
 /*****************************************************************************
  * Watchdog
  ****************************************************************************/
-static void __init kirkwood_wdt_init(void)
+void __init kirkwood_wdt_init(void)
 {
 	orion_wdt_init(kirkwood_tclk);
 }
@@ -397,7 +397,7 @@ void __init kirkwood_audio_init(void)
 /*
  * Identify device ID and revision.
  */
-static char * __init kirkwood_id(void)
+char * __init kirkwood_id(void)
 {
 	u32 dev, rev;
 
@@ -440,7 +440,7 @@ static char * __init kirkwood_id(void)
 	}
 }
 
-static void __init kirkwood_l2_init(void)
+void __init kirkwood_l2_init(void)
 {
 #ifdef CONFIG_CACHE_FEROCEON_L2_WRITETHROUGH
 	writel(readl(L2_CONFIG_REG) | L2_WRITETHROUGH, L2_CONFIG_REG);
Index: sid/arch/arm/mach-kirkwood/common.h
===================================================================
--- sid.orig/arch/arm/mach-kirkwood/common.h	2012-06-10 19:52:45.000000000 +0200
+++ sid/arch/arm/mach-kirkwood/common.h	2012-06-10 20:02:22.677136456 +0200
@@ -51,6 +51,14 @@ void kirkwood_nand_init(struct mtd_parti
 void kirkwood_nand_init_rnb(struct mtd_partition *parts, int nr_parts, int (*dev_ready)(struct mtd_info *));
 void kirkwood_audio_init(void);
 
+char *kirkwood_id(void);
+void kirkwood_l2_init(void);
+void kirkwood_rtc_init(void);
+void kirkwood_wdt_init(void);
+void kirkwood_xor0_init(void);
+void kirkwood_xor1_init(void);
+void kirkwood_crypto_init(void);
+
 extern int kirkwood_tclk;
 extern struct sys_timer kirkwood_timer;
 
